relay.go

```diff
@@ -99,15 +104,24 @@ func (r *Relayer) Relay(f *Frame) error {
 
 	f.Header.ID = destinationID
 	relayToDest.destination.Receive(f)
-	if f.isLast() {
-		r.removeRelayItem(f.Header.ID)
-	}
 	return nil
 }
 
-// Receive accepts a relayed frame.
-func (r *Relayer) Receive(f *Frame) {
-	r.conn.sendCh <- f
+// Handle all frames except messageTypeCallReq.
+func (r *Relayer) handleNonCallReq(f *Frame) error {
+	r.RLock()
+	item, ok := r.items[f.Header.ID]
+	r.RUnlock()
+	if !ok {
+		return errors.New("non-callReq for inactive ID")
+	}
+	f.Header.ID = item.remapID
+	item.destination.Receive(f)
+
+	if finishesCall(f) {
+		r.removeRelayItem(f.Header.ID)
```

Comment:
the `ID` has been modified at this point, so you're removing the wrong ID at this point